### PR TITLE
Add toggle for host networking

### PR DESCRIPTION
**Pull request by MasterPtato**
## Changes
Fixes HUB-300

### CODE REVIEW

**Comment by MasterPtato** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #390** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/390?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/rivet-gg/rivet/390?utm_source=stack-comment).

**Review by AngelOnFira — APPROVED** [review]

File: lib/bolt/config/src/ns.rs [hunk 1]
```diff
@@ -557,6 +557,8 @@ pub struct Rivet {
 	pub cdn: Cdn,
 	#[serde(default)]
 	pub billing: Option<RivetBilling>,
+	#[serde(default)]
+	pub host_networking: bool,
```
Comment (NathanFlurry) [inline lib/bolt/config/src/ns.rs:561]: nit: this is going to lead to a messy config. can we move this in to a subsection like `rivet.matchmaker.enable_host_networking`?

File: svc/pkg/game/ops/version-validate/src/lib.rs [hunk 2]
```diff
@@ -718,14 +718,17 @@ async fn handle(
 					}
 				}
 
+				let network_mode = unwrap!(LobbyRuntimeNetworkMode::from_i32(
+					docker_config.network_mode
+				));
+				let host_networking_enabled =
+					std::env::var("RIVET_HOST_NETWORKING").map_or(false, |v| v == "1");
```
Comment (NathanFlurry) [inline svc/pkg/game/ops/version-validate/src/lib.rs:725]: does this break tests?

**Review by NathanFlurry — CHANGES_REQUESTED** [review]
something's up with cargo deny. good to go otherwise.

File: svc/pkg/game/ops/version-validate/src/lib.rs (hunk 2 above)
Comment (MasterPtato) [reply svc/pkg/game/ops/version-validate/src/lib.rs:725]: Is there a case where someone is running a cluster with testing but not host networking? Should those tests be disabled if the env var isnt set?

Comment (NathanFlurry) [reply svc/pkg/game/ops/version-validate/src/lib.rs:725]: We can assert that host networking is enabled when testing is on.

**Review by NathanFlurry — CHANGES_REQUESTED** [review]

File: svc/pkg/game/ops/version-validate/src/lib.rs (hunk 2 above)
Comment (MasterPtato) [reply svc/pkg/game/ops/version-validate/src/lib.rs:725]: So in that case it shouldn't break any tests

Comment (NathanFlurry) [reply svc/pkg/game/ops/version-validate/src/lib.rs:725]: We don't assert this atm. It needs to be written.

**Review by NathanFlurry — CHANGES_REQUESTED** [review]
Needs to assert host networking enabled if tests enabled

File: svc/pkg/game/ops/version-validate/src/lib.rs (hunk 2 above)
Comment (MasterPtato) [reply svc/pkg/game/ops/version-validate/src/lib.rs:725]: Oh i see what you mean

**Review by NathanFlurry — CHANGES_REQUESTED** [review]
rivet.matchmaker.enable_host_networking still unresolved

**Comment by MasterPtato** [comment]
> rivet.matchmaker.enable_host_networking still unresolved
> 
> 


is this what you are referring to?

**Review by NathanFlurry — APPROVED** [review]

**Comment by NathanFlurry** [comment]
@MasterPtato checks failing

**Pull request merged**